drivers/UARTSerial.cpp

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#if DEVICE_SERIAL
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
```

Comment:
That was discussed in #4524 (issue is #4515). Interrupt is declared [here](https://github.com/hasnainvirk/mbed-os/blob/64a92df4554663a47c3d63348cac9c8d1ad13b1c/drivers/UARTSerial.h#L180) :)